docs/how_to/documenting_new_feature_toggles.rst

```diff
@@ -0,0 +1,94 @@
+
+Examples
+========
+
+Boilerplate template
+--------------------
+
+Copy-paste this boilerplate template to document a feature toggle::
+
+    # .. toggle_name: SOME_FEATURE_NAME
+    # .. toggle_implementation: WaffleFlag OR WaffleSwitch OR CourseWaffleFlag OR ExperimentWaffleFlag OR ConfigurationModel OR DjangoSetting
+    # .. toggle_default: True OR False
+    # .. toggle_description: Add here a detailed description of the consequences of enabling this feature toggle.
+    #   Note that all annotations can be spread over multiple lines by prefixing every line after the first by
+    #   at least two spaces, plus the leading space.
+    # .. toggle_warnings: Add here additional instructions that users should be aware of. For instance, dependency
+    #   on additional settings or feature toggles should be referenced here. If this field is not required, simply remove it.
+    # .. toggle_use_cases: temporary OR circuit_breaker OR vip OR opt_out OR opt_in OR open_edx
+    # .. toggle_creation_date: 2020-12-31
+    # .. toggle_target_removal_date: 2021-06-31 (this is required if toggle_use_cases is temporary. If not, simply remove it.)
+    # .. toggle_tickets: https://openedx.atlassian.net/browse/TICKET-xxx OR https://github.com/edx/edx-platform/pull/xxx
+    SOME_FEATURE_NAME = ...
+
+Configuration model
+-------------------
+
+Annotations can also be written in docstrings. This is particularly useful for documenting a ConfigurationModel class, for instance::
+
+    class SomeFeatureFlag(ConfigurationModel):
+        """
+        .. toggle_name: SomeFeatureFlag.enabled
+        .. toggle_implementation: ConfigurationModel
+        ...
+        """
+
+Annotation fields
+=================
+
+Most annotation fields are self-explanatory. In this section we describe in more details some of the more complex annotation fields.
+
+``toggle_implementation``
+-------------------------
+
+Must be one of the following:
+
+- "WaffleFlag", "WaffleSwitch", "CourseWaffleFlag": for objects that inherit from ``waffle_utils``' ``WaffleFlag``, ``WaffleSwitch`` or ``CourseWaffleFlag``, respectively.
+- "ExperimentWaffleFlag": this is for objects that inherit from ``lms.djangoapps.experiments.flags.ExperimentWaffleFlag``. See `this documentation page <https://openedx.atlassian.net/wiki/spaces/AC/pages/1250623700/Bucketing+users+for+an+experiment>`__ and the `reference implementation <https://github.com/edx/edx-platform/blob/master/lms/djangoapps/experiments/flags.py#L21>`__.
+- "ConfigurationModel": for objects that inherit from `config_models.models.ConfigurationModel <https://github.com/edx/django-config-models/>`__.
+- "SettingToggle", "SettingDictToggle": feature toggles can also be enabled by defining a `Django setting <https://docs.djangoproject.com/en/dev/topics/settings/>`__.
```

Comment:
Can you also add a link to the SettingToggle ADR?

Reply:
:ok_hand: